test/test_framework.py

```diff
@@ -579,19 +579,78 @@ class CustomObject:
         # Assignment and keywords like del are not supported in lambdas so functions are used instead.
         test_funcs = [(
             lambda: setattr(obj.state, 'dict', {}),
-            lambda: self.assertEqual(dict(obj.state.dict), {})
+            lambda: self.assertEqual(obj.state.dict, {})
         ), (
             lambda: obj.state.dict.update({"a": {}}),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {}})
         ), (
             lambda: obj.state.dict["a"].update({"b": "c"}),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {"b": "c"}})
         ), (
             lambda: obj.state.dict["a"].update({"d": "e"}),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c", "d": "e"}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {"b": "c", "d": "e"}})
         ), (
             lambda: obj.state.dict["a"].pop('d'),
-            lambda: self.assertEqual(dict(obj.state.dict), {"a": {"b": "c"}})
+            lambda: self.assertEqual(obj.state.dict, {"a": {"b": "c"}})
+        ), (
+            lambda: setattr(obj.state, 'list', []),
+            lambda: self.assertEqual(obj.state.list, [])
+        ), (
+            lambda: (
+                obj.state.list.append("a"),
+                obj.state.list.append(["c"]),
+            ),
+            lambda: (
+                self.assertEqual(obj.state.list[0], "a"),
+                # Nested lists need to be wrapped so that they are persisted.
+                self.assertIsInstance(obj.state.list[1], StoredList),
+                self.assertEqual(obj.state.list[1][0], "c"),
```

Comment:
This test is originally about verifying the transactional behavior on modifications, so these changes are pig-backing on it despite requiring different characteristics to verify the new behavior, just for the benefit of reusing some of its setup. Note that none of the implemented behavior cares about the commit or mutation details that are enforced in the body of the test below.

Reply:
Good point, I created a different test case and I will create a separate PR for extending the coverage of `test_mutable_types`.